plugins/net_plugin/net_plugin.cpp

```diff
@@ -228,7 +227,7 @@ namespace eosio {
       constexpr static auto stage_str( stages s );
       bool set_state( stages newstate );
       bool is_sync_required( uint32_t fork_head_block_num );
-      void request_next_chunk( std::unique_lock<std::mutex> g_sync, const connection_ptr& conn = connection_ptr() );
+      void request_next_chunk( fc::mutex *m, const connection_ptr& conn = connection_ptr() ) RELEASE(sync_mtx);
```

Comment:
Why, we are not using the parameter, I'm just passing it so the caller doesn't forget to call `g.release()`. I'd prefer not add the pointer dereferencing so it would crash if the pointer is null (which it shouldn't be).

Reply:
> Why

It's a regression of modern c++ patterns / best practices.

Reply:
OK, we resolved this bu not passing the mutex at all. Thanks Matt!